platform/darwin/src/MGLShape.mm

```diff
@@ -1,14 +1,42 @@
-#import "MGLShape.h"
+#import "MGLShape_Private.h"
+
+#import "MGLFeature_Private.h"
 
 @implementation MGLShape
 
-- (CLLocationCoordinate2D)coordinate
-{
-    [[NSException exceptionWithName:@"MGLAbstractClassException"
-                             reason:@"MGLShape is an abstract class"
-                           userInfo:nil] raise];
++ (nullable instancetype)shapeWithData:(NSData *)data encoding:(NSStringEncoding)encoding error:(NSError * _Nullable *)outError {
+    NSString *string = [[NSString alloc] initWithData:data encoding:encoding];
+    if (!string) {
+        if (outError) {
+            *outError = [NSError errorWithDomain:MGLErrorDomain code:MGLErrorCodeUnknown userInfo:nil];
+        }
+        return nil;
+    }
+    
+    try {
+        const auto geojson = mapbox::geojson::parse(string.UTF8String);
+        return MGLShapeFromGeoJSON(geojson);
+    } catch (std::runtime_error &err) {
+        if (outError) {
+            *outError = [NSError errorWithDomain:MGLErrorDomain code:MGLErrorCodeUnknown userInfo:@{
+                NSLocalizedFailureReasonErrorKey: [NSString stringWithFormat:@"%s", err.what()],
```

Comment:
I forgot to make this change before merging. Followed up in d34356ce390bc5f995e2c8e29deaa2a6bec17a10.